n2v/methods/rmks.py

```diff
@@ -6,7 +6,521 @@
+                one can run a quick WY calculation as the initial
+                guess. This can also be used to user speficified
+                initial guess by setting Da, Coca, eigvec_a.
+                3) If it's not continue, it would be expecting a
+                method name string that works for psi4. A separate psi4 calculation
+                would be performed.
+
+
+        returns:
+        ----------------------
+            all are np.ndarray of shape (num_grid_points)
+            vxc, vxchole, ebarKS, ebarWF, taup_rho_WF, taup_rho_KS
+            in eqn vxc = vxchole + ebarKS - ebarWF + taup_rho_WF - taup_rho_KS.
+            Check the original paper for the definition of each component.
+    """
+        if not self.wfn.name() in ["CIWavefunction", "RHF"]:
+            raise ValueError("Currently only supports Psi4 CI wavefunction"
+                             "inputs because Psi4 CCSD wavefunction currently "
+                             "does not support two-particle density matrices.")
+
+        if self.ref != 1:
+            raise ValueError("Currently only supports Spin-Restricted "
+                             "calculations since Spin-Unrestricted CI "
+                             "is not supported by Psi4.")
+
+        if self.guide_potential_components[0] != "hartree":
+            raise ValueError("Hartree potential is necessary as the guide potential.")
+
+        Nalpha = self.nalpha
+
+        # Preparing DFT spherical grid
+        functional = psi4.driver.dft.build_superfunctional("SVWN", restricted=True)[0]
+        self.Vpot = psi4.core.VBase.build(self.basis, functional, "RV")
+        self.Vpot.initialize()
+        self.npoints_DFT = 0
+        for blk in range(self.Vpot.nblocks()):
+            self.npoints_DFT += self.Vpot.get_block(blk).x().shape[0]
+        self.Vpot.properties()[0].set_pointers(self.wfn.Da())
+
+
+        # Preparing for WF properties
+        if self.wfn.name() == "CIWavefunction":
+            # TPDM & ERI Memory check
+            nbf = self.nbf
+            I_size = (nbf ** 4) * 8.e-9 * 2
+            numpy_memory = 2
+            memory_footprint = I_size * 1.5
+            if I_size > numpy_memory:
+                psi4.core.clean()
+                raise Exception("Estimated memory utilization (%4.2f GB) exceeds allotted memory \
+                                            limit of %4.2f GB." % (memory_footprint, numpy_memory))
+            else:
+                print("Memory taken by ERI integral matrix and 2pdm is about: %.3f GB." % memory_footprint)
+
+
+            opdm = np.array(self.wfn.get_opdm(-1,-1,"SUM",False))
+            tpdm = self.wfn.get_tpdm("SUM", True).np
+
+            Ca = self.wfn.Ca().np
+
+            mints = psi4.core.MintsHelper(self.basis)
+            I = mints.ao_eri()
+            del mints
+            # Transfer the AO ERI into MO ERI
+            I = contract("ijkl,ip,jq,kr,ls", I, Ca, Ca, Ca, Ca)
```

Comment:
I believe psi4 has mints.mo_eri(ca, ca, ca, ca). Not sure if any option is better/more efficient